Add add-homepage-1 to master

Aichnerc opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently, the ScrollToTop fix is not applied.
- Currently, the page consists of the template.
- Currently, there are no custom sections.
- Currently, there is no development warning page.


**What is the new behavior (if this is a feature change)?**
- The ScrollToTop fix has been applied.
- The page has been rebuilt by design and functionality of the [Prototype](https://github.com/aichner/Company-Homepage-Prototype).
- All the sections required for the final homepage have been added.
- A development warning has been designed. It is only visible, if not on localhost or if the website is in development mode.
- The 2020 coding style has been enforced.


**Other information**:
- This is version 1.0.0 of the Company homepage. It has already been deployed.

Aichnerc committed:
Add ScrollToTop fix

Our Scroll-To-Top fix has been implemented. This lays the 
foundation for our static use of this ReactJS application.

Aichnerc committed:
Add Hero component

The basic Hero component has been added. It still 
requires alot of tuning.

Aichnerc committed:
Add company colors

A few background colors have been changed.

Aichnerc committed:
Improve HomePage

The old template data has been replaced by the new 
component structure.

Aichnerc committed:
Improve App.js

The ScrollToTop fix has been implemented and a new 
state (globalProps) has been added.

Aichnerc committed:
Improve Routes

The globalProps have been forwareded to the HomePage 
component.

Aichnerc committed:
Add background images

The background images for the Hero component have 
been added.

LucaAllmaier committed:
Format automatically

File was formatted on save, no real changes were 
made.

LucaAllmaier committed:
Add ContactForm Section

New section for the contact form was created 
(index.js and contactForm.scss files). Built a basic 
form in the index file, scss file is still empty.

LucaAllmaier committed:
Provide the new section

The newly created section was exported from the 
sections folder and imported to the Homepage.

Aichnerc committed:
Add general onChange functions

Functions for onChange events for inputs have been 
implemented.

LucaAllmaier committed:
Extend Contact Form

Added the last onChange call and set the initial state.

LucaAllmaier committed:
Add images

Added preexisting images for the services section 
of the homepage.

LucaAllmaier committed:
Add services section

The index.jsx and sercives.scss have been created for 
the services section.

LucaAllmaier committed:
Export services

The new section has been added to the exported 
sections in this file.

LucaAllmaier committed:
Show services

The new section was imported to the homepage and 
added to the output.

LucaAllmaier committed:
Add copyright to scss file

Added the common copyright information 
to the newly created services.scss file.

LucaAllmaier committed:
Rearrange the Page order

Swapped the Cervices and ContactForm sections.

LucaAllmaier committed:
Add parallax library

The react-scroll-parallax library has been added.

LucaAllmaier committed:
Add images

2 Images for the parallax effect have been added.

LucaAllmaier committed:
Add CallToAction section

The index.jsx and callToAction.scss files have been 
added, building the CallToAction section.

LucaAllmaier committed:
Import parallax library

The installed parallax library has been importet and 
the App.js has been wrapped in a ParallaxProvider 
in order for it to work.

LucaAllmaier committed:
Export CallToAction

The newly created section is exported in order to be 
used on the homepage.

LucaAllmaier committed:
Import CallToAction

The newly created section is imported to the 
homepage and added to the output

Aichnerc committed:
Improve Hero

The image has been reintroduced. It is about to 
replaced by something better soon.

Aichnerc committed:
Improve CallToAction

Some changes have been made to improve the 
coding style and the SCSS has been implemented, 
which improved the design.

Aichnerc committed:
Improve Services

The services section has been improved by adding 
the scss and enforcing our 2020 coding style.

Aichnerc committed:
Fix CallToAction overflow

The parallax animations caused an overflow which 
caused vertical scrolling. The overflow has been 
hidden.

Aichnerc committed:
Add development page

A page has been added to display a construction 
message.

LucaAllmaier committed:
Add colors

Added two color variables from the prototype.

LucaAllmaier committed:
Add logo

Added the main_logo.svg file for the footer.

LucaAllmaier committed:
Improve footer

Improved the current footer with code from the 
prototype.

LucaAllmaier committed:
Improve footer.scss

Improved the footer.scss file in order to match the 
new footer.

Aichnerc committed:
Add .env template

The .env has been added.

Aichnerc committed:
Add Printing Page

The printing page has been added.

Aichnerc committed:
Add Branding Page

The branding page has been added.

Aichnerc committed:
Improve navigation bar

The navbar has been improved by adding design and 
functionality.

Aichnerc committed:
Improve footer

The footer has been improved by altering the content.

Aichnerc committed:
Improve Printing- and Branding Page

The two pages have been implemented in the routes.

Aichnerc committed:
Improve .gitignore

The logos of trusted companies have been excluded from 
GitHub.

Aichnerc committed:
Improve Routes

Routes for the new pages have been set.

Aichnerc committed:
Add Trusted section

The Trusted section has been added.

Aichnerc committed:
Implement new sections

The new sections have been implemented to the 
Homepage.

Aichnerc committed:
Improve Code Quality

Unused imports have been removed.

Aichnerc committed:
Add red outline button

The agency color outline button variant has been added.

Aichnerc committed:
Add dynamic MessagePage routes

The MessagePage routes are now being rendered 
dynamically.

Aichnerc committed:
Add About and Privacy

The two pages have been added.

Aichnerc committed:
Improve navbar

The navigation bar has been improved.

Aichnerc committed:
Improve code quality

The code quality has been improved.

Aichnerc committed:
Improve Footer

The content has been altered.

Aichnerc committed:
Improve CallToAction

The CallToAction section has been improved.

Aichnerc committed:
Improve Hero section

The link to the analysis page has been improved.

Aichnerc committed:
Update Services section

The buttons to learn more about each sector have been 
removed for now.

Aichnerc committed:
Improve contact form

The contact form has been improved.

pinterid commented on src/App.js:
```diff
@@ -9,26 +9,43 @@ import { BrowserRouter as Router } from "react-router-dom";
+    if (
+      window.location.hostname === "localhost" ||
+      window.location.hostname === "127.0.0.1" ||
+      isLive
```
isLive could be changes to the real company domain.

Aichnerc replied:
This should be a general switch and not be dependent on a certain domain name.

Aichnerc commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -33,6 +44,17 @@ class Navbar extends React.Component {
   };
 
+  // Get navbar mode
+  _getMode = () => {
```
Please do not start functions with `_`.
Change to: `getMode = () => {`

pinterid commented on src/components/molecules/Footer/footer.scss:
```diff
@@ -1,25 +1,136 @@
+    }
+  }
+
+  // List item styleF
```
styleF ??

pinterid commented on src/components/molecules/Footer/footer.scss:
```diff
@@ -1,25 +1,136 @@
+        }
+      }
+    }
+  }
```
Missing new line.

pinterid commented on src/components/molecules/Footer/index.jsx:
```diff
@@ -12,96 +12,306 @@ import {
+              tag="a"
+              floating
+              social="wa"
+              href="https://api.whatsapp.com/send?phone=4368120502754&text=Hey%2C%20ich%20bin%20interessiert%20an%20einem%20Angebot%20eurer%20Agentur."
```
To many chars in one row, consider using multiline string.

pinterid commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -13,15 +14,25 @@ import {
-    collapseID: "",
-  };
+  constructor(props) {
+    super(props);
```
Missing new line.

pinterid commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -33,6 +44,17 @@ class Navbar extends React.Component {
 
+  // Get navbar mode
+  _getMode = () => {
+    let opts = {};
```
Missing new line.

pinterid commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -33,6 +44,17 @@ class Navbar extends React.Component {
+      opts["dark"] = "dark";
+    } else {
+      opts["light"] = "light";
+    }
```
Missing new line.

pinterid commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -42,14 +64,25 @@ class Navbar extends React.Component {
     const { collapseID } = this.state;
+
+    // Debugging
+    console.log(location);
```
Is this console.log still required?

pinterid commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -88,9 +140,9 @@ class Navbar extends React.Component {
 /**
  * SPDX-License-Identifier: (EUPL-1.2)
- * Copyright © 2019-2020 Werbeagentur Christian Aichner
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Missing -2020.

pinterid commented on src/components/molecules/Navbar/navbar.scss:
```diff
@@ -0,0 +1,46 @@
+
+/** 
+* SPDX-License-Identifier: (EUPL-1.2)
+* Copyright © 2019 Werbeagentur Christian Aichner
```
Missing -2020

Aichnerc commented on src/components/molecules/Navbar/index.jsx:
```diff
@@ -42,14 +64,25 @@ class Navbar extends React.Component {
-        <MDBNavbar color="white" light expand="md" fixed="top" scrolling>
+        <MDBNavbar
+          color={this.props.darkMode ? "agency-dark" : "white"}
+          {...this._getMode()}
```
Alter to `...this.getMode()`

pinterid commented on src/components/organisms/sections/CallToAction/index.jsx:
```diff
@@ -0,0 +1,87 @@
+            <h2 className="gidole">
+              Es ist einfach, <strong>jetzt</strong> zu starten!
+            </h2>
+            <p>Und der Einstieg und Erstberatung sind kostenlos.</p>
```
und die Ersberatung?

Aichnerc commented on src/components/organisms/sections/CallToAction/callToAction.scss:
```diff
@@ -0,0 +1,113 @@
+#calltoaction {
+  overflow: hidden;
```
Why is this required?

pinterid commented on src/components/organisms/sections/ContactForm/contactForm.scss:
```diff
@@ -0,0 +1,54 @@
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Missing -2020.

pinterid commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -0,0 +1,214 @@
+    note: "",
+  };
+
+  componentDidMount = () => {};
```
Is this required?

Aichnerc replied:
Empty function. Please remove.

Aichnerc commented on src/components/organisms/sections/CallToAction/index.jsx:
```diff
@@ -0,0 +1,87 @@
+//> Images
+// Rocket
+import rocket from "../../../../assets/content/icons/rocket.png";
+//import rocket_pen from '../../../../assets/content/icons/rocket_pen.png';
```
The `rocket_pen` is no longer required. Please remove it.

pinterid commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -0,0 +1,214 @@
+                <MDBCardBody className="py-5">
+                  <h3 className="font-weight-bold">Kontakt aufnehmen</h3>
+                  <p className="text-muted">
+                    Du hast eine Projekt-Idee und bist Dir nicht sicher um deren
```
Most of the websites write it "Projektidee" not "Projekt-Idee".

Aichnerc commented on src/components/organisms/sections/Hero/hero.scss:
```diff
@@ -0,0 +1,118 @@
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

LucaAllmaier commented on src/components/organisms/sections/Trusted/index.jsx:
```diff
@@ -0,0 +1,103 @@
+          </h2>
+          <p className="lead">
+            Wir bieten Dir <strong>hochwertige</strong>, auf Deine Vision
+            zugeschnittene Lösungen in sämtlichen Bereichen des Marketing.
```
des Marketings?

Aichnerc replied:
Damn. I just changed that today. 👀😂

Aichnerc commented on src/components/organisms/sections/Hero/index.jsx:
```diff
@@ -0,0 +1,147 @@
+    let min = Math.ceil(10);
+    let max = Math.floor(100);
+    let arr = [1, 2, 3, 4, 5, 6, 7, 8];
+
```
Remove the space

pinterid replied:
Remove new line.

Aichnerc commented on src/components/organisms/sections/Services/index.jsx:
```diff
@@ -0,0 +1,107 @@
+      },
+    },
+  ],
+};
```
This data could be loaded in from a seperate file.

pinterid commented on src/components/organisms/sections/ContactForm/index.jsx:
```diff
@@ -0,0 +1,214 @@
+                        Du kannst uns täglich von 9:00 bis 18:00 telefonisch
+                        unter
+                        <br />
+                        <strong>+43 681 205 027 54</strong> erreichen.
```
Add a link on the tel number?

Aichnerc commented on src/components/organisms/sections/Trusted/index.jsx:
```diff
@@ -0,0 +1,103 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

Aichnerc commented on src/components/organisms/sections/Trusted/trusted.scss:
```diff
@@ -0,0 +1,11 @@
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

pinterid commented on src/components/organisms/sections/Hero/hero.scss:
```diff
@@ -0,0 +1,118 @@
+  &.dark {
+    background: linear-gradient(to top, $dark 0%, darken($dark, 10%) 100%);
+    color: white;
+  }
```
Missing new line.

pinterid commented on src/components/organisms/sections/Hero/index.jsx:
```diff
@@ -0,0 +1,147 @@
+  MDBIcon,
+  MDBCard,
+  MDBCardBody,
+} from "mdbreact";
```
Should be moved above MDB.

Aichnerc commented on src/components/pages/BrandingPage/branding.scss:
```diff
@@ -0,0 +1,81 @@
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

Aichnerc commented on src/components/pages/BrandingPage/index.jsx:
```diff
@@ -0,0 +1,367 @@
+import weddingsLogo from '../../../assets/content/weddings.png';
+
+class BrandingPage extends React.Component {
+
```
Remove the new line

Aichnerc commented on src/components/pages/BrandingPage/index.jsx:
```diff
@@ -0,0 +1,367 @@
+  }
+
+  downloadImage = (type, format) => {
+
```
Remove the new line

pinterid replied:
Remove new line.

LucaAllmaier commented on src/components/pages/BrandingPage/index.jsx:
```diff
@@ -0,0 +1,367 @@
+// Logo
+import { ReactComponent as Logo } from  '../../../assets/content/logo_full.svg';
+// Wedding logo
+import weddingsLogo from '../../../assets/content/weddings.png';
```
use " "

Aichnerc commented on src/components/pages/BrandingPage/index.jsx:
```diff
@@ -0,0 +1,367 @@
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

Aichnerc commented on src/components/pages/DevelopmentPage/index.jsx:
```diff
@@ -0,0 +1,64 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

Aichnerc commented on src/components/pages/PrintingPage/index.jsx:
```diff
@@ -0,0 +1,258 @@
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

Aichnerc commented on src/components/pages/PrintingPage/printing.scss:
```diff
@@ -0,0 +1,93 @@
+
+/** 
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © 2019 Werbeagentur Christian Aichner
```
Copyright should be 2019-2020.

pinterid replied:
Missing -2020.

Aichnerc commented on src/index.js:
```diff
@@ -4,6 +4,10 @@ import React from "react";
 
+//> Additional libraries
+// Parallax
+import { ParallaxProvider } from 'react-scroll-parallax';
```
Use double quotes instead of single quotes.

Aichnerc commented on src/utilities/_variables.scss:
```diff
@@ -2,6 +2,8 @@ $dark: #212529;
 $green: #77bd43;
 $pink: #ea4aaa;
+$agencyDark: #212529;
+$agencyRed: #f61a42;
```
Unify red and agencyRed.

Aichnerc reviewed (commented):
There are alot of issues to be fixed. I wish myself happy fixing.

sweggerat reviewed (approved):
looks bad

pinterid commented on src/components/pages/BrandingPage/index.jsx:
```diff
@@ -0,0 +1,367 @@
+    }
+
+    //> Temp output of the file URL
+    console.log(options);
```
Is this still required?

pinterid commented on src/components/pages/BrandingPage/index.jsx:
```diff
@@ -0,0 +1,367 @@
+  }
+
+  render() {
+
```
Remove new line.

pinterid commented on src/components/pages/PrintingPage/printing.scss:
```diff
@@ -0,0 +1,93 @@
+        right: auto;
+      }
+    }
+
```
Remove new line.

pinterid reviewed (changes requested):
Should be good to merge after a few changes.

LucaAllmaier reviewed (commented):
Looks good, just a few comments.

Aichnerc committed:
Improve code quality

The code quality has been improved by the request of 
@Aichnerc, @pinterid and @LucaAllmaier.

Aichnerc reviewed (commented):
Looks good.

pinterid reviewed (approved):
Looks good to merge.
